simplic-corelib/Simplic.CoreLib/Data/Numeric.cs

```diff
@@ -0,0 +1,318 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Simplic.Data
+{
+    /// <summary>
+    /// A numeric data type which accepts all kinds of numeric types
+    /// </summary>
+    public struct Numeric : IComparable, IConvertible, IFormattable
+    {
+        private const int N = 8;
+        private readonly double _value;
```

Comment:
Maybe saving the original object here is better?

Reply:
it is then hard to convert to a common type when we are trying to multiply. this way everything is in double range.